osquery/events/eventfactory.cpp

```diff
@@ -427,6 +428,10 @@ void EventFactory::delay() {
     return;
   }
 
+  if (shutdownRequested()) {
```

Comment:
Could you combine this check and the check in `attachEvents` into a single check within `Initializer::start`?